packages/web-config-server/src/apiV1/dataBuilders/generic/percentage/percentagesOfValueCounts.js

```diff
@@ -138,10 +138,17 @@ export class PercentagesOfValueCountsBuilder extends DataBuilder {
 
     const allResults = numeratorResults;
 
+    const numeratorResultsMap = {};
+    numeratorResults.forEach(analytic => {
+      const { organisationUnit, dataElement, value, period } = analytic;
+      numeratorResultsMap[`${organisationUnit}|${dataElement}|${value}|${period}`] = analytic;
+    });
+
     // Hack to make sure that there are no duplicated analytics returned to count twice.
     // Would like to have { denominatorResults, numeratorResults }, but can't because of how DataPerPeriodBuilder works
     denominatorResults.forEach(analytic => {
-      if (!allResults.find(otherAnalytic => isEqual(analytic, otherAnalytic))) {
```

Comment:
This was way to slow before especially for reports that have around 14k of analytics in Laos schools and blocked the thread. Is this a good way to improve the performance?

@Alastair-L @rohan-bes

Reply:
Hmmm the only issue I can see with this is that if it were possible to have duplicate analytics within the denominator results, then this new code could allow for those duplicates to be added to `allResults`

However, I don't believe that is possible... @Alastair-L maybe you know?

Reply:
Looks like it at least avoids the double loop that I introduced 👍. You could maybe store those keys in a set if you wanted to improve memory usage, since the values are never used?

"numeratorResultsMap[`${organisationUnit}|${dataElement}|${value}|${period}\`] = analytic;"
=> "numeratorResultsSet.add(`${organisationUnit}|${dataElement}|${value}|${period}`)"

Reply:
Oops! @rohan-bes, sorry I didn't see your comment before approving! That isn't possible to my knowledge, but I'm not 100% sure